src/components/IconButton/IconButton.tsx

```diff
@@ -1,28 +1,66 @@
 import React, { FC, JSXElementConstructor } from 'react';
 
 import classNames from 'classnames';
-import styles from './IconButton.module.scss';
 import { IconProps } from '../Icons/Icons.types';
 
+import './IconButton.scss';
+
 export interface IconButtonProperties {
   className?: string;
   onClick: () => void;
 
   Icon: JSXElementConstructor<IconProps>;
   label?: string;
-  size?: string | number;
+  size?: string | number | 'large' | 'small' | 'x-small';
   isFilled?: boolean;
+  variant?: 'primary' | 'secondary' | 'tertiary';
+  color?: 'primary' | 'red' | 'greyscale';
+  isDisabled?: boolean;
 }
 
-export const IconButton: FC<IconButtonProperties> = ({ Icon, onClick, className, label, size, isFilled }) => {
+export const IconButton: FC<IconButtonProperties> = ({
+  Icon,
+  onClick,
+  className,
+  label,
+  size,
+  isFilled,
+  variant,
+  color,
+  isDisabled
+}) => {
   function handleClick(event: React.MouseEvent<HTMLButtonElement>) {
     event.stopPropagation();
     onClick();
   }
 
+  const getSize = () => {
+    if (size && ['large', 'small', 'x-small'].includes(size.toString())) {
+      switch (size) {
+        case 'large':
+          return 40;
+        case 'small':
+          return 32;
+        case 'x-small':
+          return 24;
+      }
+    }
+
+    return size;
+  };
```

Comment:
nit: move this function outside of the component, e.g.

```ts
const IconButton = () => { ... };

const getSize = (size: IconButtonProperies['size']) => {
    if (size && ['large', 'small', 'x-small'].includes(size.toString())) {
      switch (size) {
        case 'large':
          return 40;
        case 'small':
          return 32;
        case 'x-small':
          return 24;
      }
    }

    return size;
  };
```

Reply:
done.